### DISCUSSION TITLE
Problem with jekyll permalink during deploy.

### DISCUSSION THREAD

**Discussion by muhashfar** [vercel/vercel#5352] [Help] [answered]
The issue is I already put "permalink: /icons" to front matter of my page. But its is not showing my page instead of showing custom 404 directory. But the page is visible when add ".html".
Kindly anyone please help.

**Comment by bigsee** [comment, marked as answer]
Did you ever find an answer to this? I have the same issue. adding .html works fine but without I get 404 for all pages.

**Reply by lelong88** [reply]
I had the same issue and found a solution here: https://vercel.com/docs/project-configuration

You'll need to add a new file `vercel.json` to root directory of your vercel project with following content:
`{
  "cleanUrls": true
}`

This will instruct vercel server to serve jekyll generated html files without the `.html` extension

**Comment by amyegan** [comment]
Can you share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example)? I'm not very familiar with Jekyll, but I'm happy to do some digging
